GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -596,6 +596,9 @@ private void RefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs)
                 return;
             }
 
+            // Process events e.g. from closing FormProcess, particularly after rebase / conflict resolution actions
+            Application.DoEvents();
```

Comment:
This is a "smell", meaning we're doing something wrong...
Can `Invoke` instead?

Reply:
> This is a "smell", meaning we're doing something wrong...

Yes, but what?

> Can `Invoke` instead?

No, `Invoke((Action)(() => RevisionGrid.PerformRefreshRevisions(getRefs)));` does not suffice.
And this is no surprise because `UICommands_PostRepositoryChanged` is executed in the UI thread already.

Reply:
The calling was changed in #9897. `RefreshRevisions()` is no longer async, but calculations is in async anyway, except for #9899 if artificial that is worse (still issues before and still not async in other situations) after this - I missed that, the calling structure was too complicated.

Reply:
> And this is no surprise because `UICommands_PostRepositoryChanged` is executed in the UI thread already.

Perhaps this is the underlying problem - we use this event for I/O, and not just for the UI updates. 
I'm not asking for a change, just making an observation.

Reply:
💭 thinking out loud - may be we need to have two events - one is fired on a background thread for I/O (`UICommands_PostRepositoryChanged`) and another one (`DataRefreshed`) - one the UI thread for the UI updates.

Reply:
This PR can wait until after #9888 and perhaps other follow-ups, which might even make it obsolete.
(I just reproduced the addressed issue many times when testing #9907 and had a look.)